src/sql/workbench/contrib/charts/browser/chartView.ts

```diff
@@ -389,8 +389,9 @@ export class ChartView extends Disposable implements IPanelView {
 				numberInput.setAriaLabel(option.label);
 				numberInput.value = value || '';
 				numberInput.onDidChange(e => {
-					if (this._options[entry] !== Number(e)) {
-						(this._options[entry] as any) = Number(e);
+					if (this._options[entry] !== e) {
+						// When user clears the input box, the value we get from the input box will be empty string.
+						(this._options[entry] as any) = e === '' ? undefined : Number(e);
```

Comment:
nit: this is not the most clear statement to read, with the tertiary operator and the cast.  maybe wrapping `e === '' ? undefined : Number(e)` in `( )` would help a little.